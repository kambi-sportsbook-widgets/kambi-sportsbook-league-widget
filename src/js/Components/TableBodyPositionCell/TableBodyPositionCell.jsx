import React, { PropTypes } from 'react';
import { translationModule } from 'kambi-widget-core-library';
import styles from './TableBodyPositionCell.scss';

const t = translationModule.getTranslation.bind(translationModule);

const TableBodyPositionCell = ({ children }) => {
   return (
      <td className={styles.general} title={t('Position')}>
         {children}
      </td>
   );
};

TableBodyPositionCell.propTypes = {
   /**
    * Inner element - PositionIndicator
    */
   children: PropTypes.element.isRequired
};

export default TableBodyPositionCell;
