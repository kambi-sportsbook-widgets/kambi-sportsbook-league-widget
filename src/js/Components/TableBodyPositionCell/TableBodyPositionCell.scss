:local(.general) {
   white-space: nowrap;
   width: 54px;
   padding-left: 16px;
}
