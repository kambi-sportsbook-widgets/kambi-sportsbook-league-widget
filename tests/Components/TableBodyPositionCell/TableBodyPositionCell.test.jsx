/* eslint-env jest */
import React, { Children } from 'react';
import TableBodyPositionCell from '../../../src/js/Components/TableBodyPositionCell/TableBodyPositionCell';
import ReactTestRenderer from 'react-test-renderer';

jest.mock('kambi-widget-core-library', () => ({
   translationModule: {
      getTranslation: (key) => `Translated: "${key}"`
   }
}));

describe('TableBodyPositionCell DOM rendering', () => {

   it('renders correctly', () => {
      const tree = ReactTestRenderer.create(
         <TableBodyPositionCell>
            <div>Test</div>
         </TableBodyPositionCell>
      ).toJSON();

      expect(tree).toMatchSnapshot();
   });

});
