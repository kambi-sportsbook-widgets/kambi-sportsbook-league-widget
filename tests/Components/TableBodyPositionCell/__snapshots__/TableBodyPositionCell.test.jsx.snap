// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBodyPositionCell DOM rendering renders correctly 1`] = `
<td
  className="general"
  title="Translated: \\"Position\\""
>
  <div>
    Test
  </div>
</td>
`;
